### ISSUE TITLE
ngeo is too dependency heavy

### ISSUE THREAD

**Issue by tsauerwein** [camptocamp/ngeo#2004]
When using ngeo in a project, I am forced to include all dependencies of ngeo, also when I am not using the service or directive that actually requires a certain dependency.

The ngeo module is specified with:

    ngeo.module = angular.module('ngeo', ['gettext', 'ui.date', 'floatThead']);

https://github.com/camptocamp/ngeo/blob/master/src/ngeo.js

This means to use ngeo, I have to install gettext, ui.date and also floatThead, even when I don't need it. The same applies to the [gmf module](https://github.com/camptocamp/ngeo/blob/master/contribs/gmf/src/gmf.js).

It should be possible to use certain components of ngeo without having to install the dependencies of all components.

Additionally, because the dependencies are listed as `devDependencies`, they have to be listed again as dependency in my project.

**Comment by gberaudo** [comment]
Currently, the ngeo code only uses one module, which depends on all the ngeo code.
The gmf module depends on all the gmf code + the ngeo code + extra dependencies.

If we had a hierarchy of fine grained modules, every application could depend on the specific subset that it actually needs. We should refactor ngeo to achieve this property if we want it to be used in more applications.

I discussed this with @sbrunner. Here is our idea:
- create a `src/modules` directory;
- create a subdirectory per module, eg: src/modules/map:
    ./module.js -> the map module definition
    ./directive1.js and ./partial1.html -> a directive for this module and its partial
    ./service1.js -> a service for this module
- create a top level all module for what is currently the ngeo module: src/modules/all
     ./module.js -> goog.require of all other modules
  
The directives/services/.. would be registered in angular by their module and not in the directive code like it is done currently. Eg: for https://github.com/camptocamp/ngeo/blob/master/src/directives/map.js (see the bottom).
modules/map/maps.js: remove `ngeo.module.directive('ngeoMap', ngeo.mapDirective);`
modules/map/module.js: contain something like:
```
goog.provide('ngeo.modules.map.module');
goog.require('ngeo.mapDirective');
ngeo.modules.map.module = angular.module('mapModule', []);
ngeo.modules.map.module.module.directive('ngeoMap', ngeo.mapDirective);
```

**Comment by ybolognini** [comment]
Corrected in 2.3.
